### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by the-faizmohammad**
This Pull request contains following changes.
- Setup the rubocop linter tools.
- As I faced an issue installing the required gems using bundle install, I used a different approach to install gems and included the .gitignore file to ignore the packages.
- Created three different script files for the classes Person, Student and Teacher.
- Created the class Person with instance variables - id, name, age and parent_permission.
- Used the attr_reader and attr_accessor for variables getters and setters.
- Included a private method of_age? and another public method can_use_services? in the class.
- Created the Student subclass deriving from the Person with instance variable - @classroom. also included the play_hooky method.
- Created the Teacher subclass also deriving from the Person with instance variable - @specialiazation and a derived method overriding the can_use_services? result.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+def can_use_services?
+  of_age? || @parent_permission
+end
+
+  private
+
+def of_age?
+  @age >= 18
+end
```
Comment (LionRouge1) [inline person.rb]: - [ ] Kindly create a class `Person` and move all the code in this page into the class `Person`

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+def initialize(id, name: 'Unknown', age: 0, parent_permission: true)
+  @id = id
+  @name = name
+  @age = age
+  @parent_permission = parent_permission
+end
+
+attr_writer :name
+
+attr_writer :age
```
Comment (LionRouge1) [inline person.rb]: - [ ] According to the project requirement we should create a getter and setter for the variables `@name` and `@age` and only a getter for `@id`.  Therefore I suggest you add them to follow the project requirements.
Screenshot from 2023-12-05 16-47-13
```rb
  attr_accessor :name, :age
  attr_reader :id
```

File: README.md [hunk 3]
```diff
@@ -1 +1,139 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
+
```
Comment (LionRouge1) [inline README.md:138]: - [ ] Your license file has no content, kindly make  sure you add the license content to make your project look more professional

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @the-faizmohammad, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require_relative 'Person'
+
+class Teacher < Person
+  def initialize(id, specialization:, name: 'Unknown', age: 0, parent_permission: true)
+    super(id, name: name, age: age, parent_permission: parent_permission)
+    @specialization = specialization
+  end
```
Comment (julie-ify) [inline teacher.rb]: - [ ] After creating the Person class you still need to ensure the classes that inherit from it are structured correctly. There is no `id`, `age` is not optional, the order of the parameters should be followed, and the way of assigning the parameter should match the Person class.

  ```
  def initialize(specialization, age, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @specialization = specialization
  end
  ```

File: student.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+require_relative 'Person'
+
+class Student < Person
+  def initialize(id, classroom:, name: 'Unknown', age: 0, parent_permission: true)
+    super(id, name: name, age: age, parent_permission: parent_permission)
+    @classroom = classroom
+  end
```
Comment (julie-ify) [inline student.rb]: - [ ] After creating the Person class you still need to ensure the classes that inherit from it are structured correctly. There is no `id`, `age` is not optional, the order of the parameters should be followed, and the way of assign the parameter should match the Person class.

  ```
  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @classroom = classroom
  end
  ```

File: README.md [hunk 6]
```diff
@@ -1 +1,139 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (julie-ify) [inline README.md:137]: - [ ] Your license file has no content as highlighted by the previous reviewer. Kindly use the template and customize it.

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @the-faizmohammad,

Good job so far!

## Highlights!
- Good code organization and classes ✔️
- Descriptive README ✔️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @the-faizmohammad 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All your classes are now  correct :heavy_check_mark: 
- Broken link is fixed :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**